task/git-cli/0.3/git-cli.yaml

```diff
@@ -0,0 +1,139 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: git-cli
+  labels:
+    app.kubernetes.io/version: "0.3"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.21.0"
+    tekton.dev/categories: Git
+    tekton.dev/tags: git
+    tekton.dev/displayName: "git cli"
+    tekton.dev/platforms: "linux/amd64,linux/s390x,linux/ppc64le"
+spec:
+  description: >-
+    This task can be used to perform git operations.
+
+    Git command that needs to be run can be passed as a script to
+    the task. This task needs authentication to git in order to push
+    after the git operation.
+
+  workspaces:
+    - name: source
+      description: A workspace that contains the fetched git repository.
+
+    - name: input
+      description: A workspace that contains file that needs to be added to git.
```

Comment:
can we make this as optional as it may not be necessary that user may bind a workspace containing the file which needs to be committed to git?

Reply:
Great idea!

Reply:
I've added this explanation
```
* **input**: An optional workspace that contains the files that need to be added to git. You can
    access the workspace from your script using `$(workspaces.input.path)`, for instance:

      cp $(workspaces.input.path)/file_that_i_want .
      git add file_that_i_want
      # etc
```
but I'm not actually sure if that is correct 😅. _Is_ that the correct order of evaluation? Or will the reference inside the script not be evaluated anymore?

Reply:
yeah this will work.